### PR TITLE
Make ClusterNetworkAddressConfigProvider and co internal

### LINKED ISSUE #1381: Move ClusterNetworkAddressConfigProvider and associated deprecated classes to an internal module
See #646, in 0.3.0 we deprecated:

ClusterNetworkAddressConfigProvider
ClusterNetworkAddressConfigProviderContributor
HostPort

because we want to move them to an internal module until we decide they are a public API

They are eligible to be moved in 0.7.0

### PR DESCRIPTION
**Pull request by robobario**
### Type of change

- Refactoring

### Description

Closes #1381

These interfaces may become a supported pluggable interface one day, but for now we want to reduce the API surface area we need to support. They have been deprecated long enough that we can make the breaking change required as per our deprecation policy.

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Review performance test results. Ensure that any degradations to performance numbers are understood and justified.
- [ ] Make sure all Sonarcloud warnings are addressed or are justifiably ignored.
- [ ] Update documentation
- [ ] Reference relevant issue(s) and close them after merging
- [ ] For user facing changes, update CHANGELOG.md (remember to include changes affecting the API of the test artefacts too).

### CODE REVIEW

File: kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/service/HostPort.java
Comment (robobario) [inline kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/service/HostPort.java:1]: sonar doesn't like the ipv6 regex

> Make sure the regex used here, which is vulnerable to polynomial runtime due to backtracking, cannot lead to denial of service.

Maybe we should merge this with a suppress and make an issue to follow up.

**Review by k-wall — APPROVED** [review]
lgtm

File: kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/service/HostPort.java
Comment (SamBarker) [reply kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/service/HostPort.java:1]: Do we actually need a follow up issue? We are sure that we want an IPv6 address and that its not exposed to arbitrary input only that from the config file so I'm happy to merge with a suppression and nothing else.

**Review by SamBarker — APPROVED** [review]
LGTM thanks @robobario

**Pull request merged**